profile/src/v100/entity_security_state/unsecured_entity_control.rs

```diff
@@ -0,0 +1,154 @@
+use crate::prelude::*;
+
+/// Basic security control of an unsecured entity. When said entity
+/// is "securified" it will no longer be controlled by this `UnsecuredEntityControl`
+/// but rather by an `AccessControl`. It is a name space holding the
+/// single factor instance which was used to create
+#[derive(
+    Serialize, Deserialize, Clone, Debug, PartialEq, Eq, Hash, uniffi::Record,
+)]
+#[serde(rename_all = "camelCase")]
+pub struct UnsecuredEntityControl {
+    // /// The factor instance which was used to create this unsecured entity, which
+    // /// also controls this entity and is used for signing transactions.
+    pub transaction_signing: HierarchicalDeterministicFactorInstance,
+
+    /// The factor instance which can be used for ROLA.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub authentication_signing: Option<HierarchicalDeterministicFactorInstance>,
+}
+
+impl UnsecuredEntityControl {
+    pub fn with_entity_creating_factor_instance<T>(
+        entity_creating_factor_instance: HDFactorInstanceTransactionSigning<T>,
+    ) -> Self
+    where
+        T: IsEntityPath + Clone,
+    {
+        Self {
+            transaction_signing: entity_creating_factor_instance.into(),
+            authentication_signing: None,
+        }
+    }
+
+    #[cfg(not(tarpaulin_include))] // false negative
+    pub fn new(
+        transaction_signing: HierarchicalDeterministicFactorInstance,
+        authentication_signing: Option<HierarchicalDeterministicFactorInstance>,
+    ) -> Result<Self> {
+        if let Some(auth) = &authentication_signing {
```

Comment:
is it possible to chain ifs in rust, thus reducing one level on indentation? #PossibleInSwift

Reply:
Maybe by using `match` would be an alternative to if-let, probably something like:
```rust
match &authentication_signing  {
  Some(auth) if auth.key_kind() != Some(...) {}
}
```